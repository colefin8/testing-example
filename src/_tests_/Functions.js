module.exports = {
  generateColor: () => {
    const letters = "0123456789ABCDEF";
    let newColor = "#";
    for (let i = 0; i < 6; i++) {
      newColor += letters[Math.floor(Math.random() * 16)];
    }
    //#usually set color to state
    //setColor(newColor);
    //#for this test just return color
    return newColor;
  },
};
