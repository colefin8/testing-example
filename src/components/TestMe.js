import React from "react";

const TestMe = (props) => {
  return (
    <div
      style={{
        margin: "300px auto",
        height: "300px",
        width: "300px",
        border: "1px solid black",
        backgroundColor: "white",
        color: "black",
        fontSize: '48px'
      }}
    >
      {props.name}
    </div>
  );
};

export default TestMe;
