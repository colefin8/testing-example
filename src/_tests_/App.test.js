import React from "react";
import { generateColor } from "./Functions";
import renderer from "react-test-renderer";
import TestMe from "../components/TestMe";

//# when to use act()? When writing UI tests, tasks like rendering, user events, or data fetching can be considered as “units” of interaction with a user interface. React provides a helper called act() that makes sure all updates related to these “units” have been processed and applied to the DOM before you make any assertions:

describe("unit tests for App.js", () => {
  let testColor = generateColor();
  const possibleValues = "0123456789ABCDEF";

  test("random color generator only generates valid hex codes (starts with #)", () => {
    expect(testColor[0]).toBe("#");
  });

  test("random color generator only generates valid hex codes (string of 7 chars)", () => {
    expect(testColor.length).toBe(7);
  });

  test("random color generator only generates valid hex codes (6 random values are between 0-9, or A-F)", () => {
    //#technically has 6 expect statements in one test
    for (let i = 1; i < 7; i++) {
      expect(possibleValues).toContain(testColor[i]);
    }
  });

  test("random color generator creates a new random color each time", () => {
    const newColor = generateColor();
    expect(newColor).not.toBe(testColor);
  });

});

describe("component tests for TestMe component", () => {
  let name = "cole";
  const component = renderer.create(<TestMe name={name} />);
  let tree = component.toJSON();

  test("component renders the same each time if given same input", () => {
    expect(tree).toMatchSnapshot();
  });

  test("component displays name prop", () => {
    expect(tree.children[0]).toBe(name);
  });
});
