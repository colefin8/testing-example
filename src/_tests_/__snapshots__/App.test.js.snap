// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component tests for TestMe component component renders the same each time if given same input 1`] = `
<div
  style={
    Object {
      "backgroundColor": "white",
      "border": "1px solid black",
      "color": "black",
      "fontSize": "48px",
      "height": "300px",
      "margin": "300px auto",
      "width": "300px",
    }
  }
>
  cole
</div>
`;
